### ISSUE TITLE
Sonarqube 7.6 support

### ISSUE THREAD

**Issue by pethers** [jensgerdes/sonar-pmd#81]
org.sonar.api.profiles.RulesProfile has been removed.

`[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project cia-all: org.sonar.plugins.pmd.PmdSensor has unsatisfied dependency 'class org.sonar.api.profiles.RulesProfile' for constructor 'public org.sonar.plugins.pmd.PmdSensor(org.sonar.api.profiles.RulesProfile,org.sonar.plugins.pmd.PmdExecutor,org.sonar.plugins.pmd.PmdViolationRecorder,org.sonar.api.batch.fs.FileSystem)' from org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@634d56eb:57<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@790f4933:379<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@6f6998b6:47<| ->`

**Comment by vladonemo** [comment]
Just faced the same. The only possible workaround is to uninstall PMD plugin from SonarQube.

**Comment by ediri** [comment]
`Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.6.0.1398:sonar (default-cli) on project backend-example: org.sonar.plugins.pmd.PmdSensor has unsatisfied dependency 'class org.sonar.api.profiles.RulesProfile' for constructor 'public org.sonar.plugins.pmd.PmdSensor(org.sonar.api.profiles.RulesProfile,org.sonar.plugins.pmd.PmdExecutor,org.sonar.plugins.pmd.PmdViolationRecorder,org.sonar.api.batch.fs.FileSystem)' from org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@44d7e24:132<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@1f1ff879:524<[Immutable]:org.sonar.core.platform.ComponentContainer$ExtendedDefaultPicoContainer@6736f40f:65<| -> [Help 1]`

We had the same issue with findbugs. And the team there fixed it, quite quick,  as it was a small change  in in the code.

Follow https://github.com/spotbugs/sonar-findbugs/pull/244 for more details

Hope it helps you. For the moment Sonarqube 7.6 is dead as we need the PMD Plugin...

**Comment by ediri** [comment]
Hi,

are threre any plans für updating the plugin?

**Comment by aepfli** [comment]
+1

**Comment by smiro21** [comment]
+1

**Comment by CelesteGH** [comment]
+1

**Comment by jensgerdes** [comment]
Hi @pethers 

Thank you for the issue. I can confirm the reported problem and will check on that as soon as possible.
Unfortunately, my free time is very limited at the moment - which means if this is a bigger change, I won't be able to fix this within the next days.

Pull requests are welcome.

**Comment by jensgerdes** [comment]
The problem seems to be that _RulesProfile_ has been removed from ScannerSide, remaining in ServerSide scope. At compile time everything is fine because the class is still there. However, at runtime, components running in ScannerSide scope can not use it anymore.

See: https://community.sonarsource.com/t/unsatisfied-dependency-in-sq-7-6/6244/4

The problem here is that Sonar-PMD does not clearly separate ServerSide from ScannerSide components. We'll have to separate both from each other so that ServerSide components keep using _RulesProfile_ whereas ScannerSide components switch to _ActiveRules_.

**Comment by Vampire** [comment]
Besides the findbugs fix that was linked above, you can also have a look at https://github.com/checkstyle/sonar-checkstyle/pull/193, as the checkstyle plugin also suffers from the same issue.

**Comment by jensgerdes** [comment]
I have fixed the problem. However, I won't create a release until I have thoroughly tested my changes.
Feel free to check if this fixes your problems: 
https://github.com/jensgerdes/sonar-pmd/releases/tag/3.2.0-beta

**Comment by pethers** [comment]
@jensgerdes thanks for the fix, will test it later tonight.

**Comment by pethers** [comment]
Have run the latest snapshot build on http://192.168.1.15:9000/sonar/dashboard?id=com.hack23.cia%3Acia-all , build passes so happy with the result :)

**Comment by ediri** [comment]
any ideas, when the plugin will be avail. in the marketplace?

**Comment by aepfli** [comment]
sidenote: we also have the latest snapshot running on our live and test instances for over a week now, without any issues.

**Comment by jensgerdes** [comment]
Fair question. I will try to release it within this week.

**Comment by ediri** [comment]
@jensgerdes : appreciating your great work for the sonar community, but did you had the time to release the plugin? Thx

**Comment by ceastman-ibm** [comment]
@jensgerdes just checking on the release of this fix? We are trying to install sonarqube 7.6 but are currently blocked by this.

**Comment by agabrys** [comment]
FYI: 3.2.0-SNAPSHOT version works correctly on SonarQube Community Edition 7.7 👍

**Comment by patsevanton** [comment]
![Снимок экрана от 2019-03-30 20-30-53](https://user-images.githubusercontent.com/10828883/55277497-08ed9d80-532b-11e9-948c-a8384574285b.png)

![Снимок экрана от 2019-03-30 20-31-03](https://user-images.githubusercontent.com/10828883/55277498-0ab76100-532b-11e9-9db2-d7fb2e020bba.png)

**Comment by jensgerdes** [comment]
Oops... yes, you are right. I'll fix the description

**Comment by jensgerdes** [comment]
I have [requested](https://community.sonarsource.com/t/new-release-sonar-pmd-plugin-3-2-0/8313) the release of Sonar-PMD 3.2.0 in the market place.
Sorry for the delay. I've just built a house and had not much time to take care of this project...

**Comment by mellebie** [comment]
Has this been made available in marketplace yet?  Facing same issue (RulesProfile) as others from past, when using Sonarqube 7.6 and pitest

**Comment by jensgerdes** [comment]
> Has this been made available in marketplace yet? Facing same issue (RulesProfile) as others from past, when using Sonarqube 7.6 and pitest

Hi @mellebie 
Yes, the release has been made available. See: https://docs.sonarqube.org/display/PLUG/Plugin+Version+Matrix

If you're still experiencing problems, please file in another issue along with:
* Sonar-PMD version
* Stack trace
* optionally the source code

**Comment by mellebie** [comment]
@jensgerdes , apologies for my ignorance.  Unless I am missing something, from link provided I do not see anything to download.  Where exactly should I be going, and what steps to install?  Again, I apologize.

**Comment by mellebie** [comment]
I got it to work. I installed the latest sonar-pmd from sonarqube marketplace, and now the error is resolved

**Comment by jensgerdes** [comment]
> I got it to work. I installed the latest sonar-pmd from sonarqube marketplace, and now the error is resolved

Excellent. :-)
